### ISSUE TITLE
Async methods causing problems

### ISSUE THREAD

**Issue by ghost** [uchicago-mobi/2016-Winter-Forum#87]
None of my (and, perhaps for some of my peers, our) other classes have provided a lot of detail on how to handle concurrency with asynchronous web requests. It's a challenge to synthesize the available online material on the topic (which could fill its own library) and there's a number of errors in the current project that are strictly due to concurrency issues.

Like, my circle view controller can't receive the request fast enough to output the right information to labels. The only way I can think to do it is a Busy Waiting or a spinlock or something because I don't really understand how semaphores & threading work in iOS. Not sure I'll have it resolved by 5:30, either.

A primer on this topic would be very, very helpful.

**Comment by shuaib88** [comment]
I agree I'd love more info on the topic, 

but check out the #78 i initially had those same problems but the async_dispatch function solved those for me in every case. I don't know exactly how it works, but I think it doesn't execute the code block inside it until the api call has returned.

**Comment by cnfshandson** [comment]
Do you get a lot UI background thread update warnings?  
I had the same issues, but fixed with the following method. 
dispatch_async(dispatch_get_main_queue(), {

```
                })
```

**Comment by ghost** [comment]
@shuaib88 Thanks for the hint, but I'm not sure what I should be wrapping in the async_dispatch block? I tried wrapping a call to reloadData like that but then my table view doesn't load at all...

@cnfshandson No UI background thread update warnings that I can see.

**Comment by aaizuss** [comment]
@shuaib88 Did you put `synch_dispatch` at the end of `viewDidLoad`? That's what I did in my other classes, but  when I put my code for counting the total issues in my CircleViewController, nothing happens (even when I switch back and forth between tabs). I've also tried putting the code in `viewDidAppear`, which updates the totals, but of course it continues counting and adding to the totals every time I open the tab.

**Comment by shuaib88** [comment]
Is this for your issues controller? It's tough to say without looking at your code. For issue controller I'm only wrapping self.tableView.reloadData() 

when you add that Is it not loading or crashing? I specified values for the number of rows etc.. in case there is a nil.

**Comment by aaizuss** [comment]
@shuaib88 No, my Issue controller (and any controller with tables) works fine. I'm just talking about the circle view controller.

-- EDIT: I figured it out. I just needed to put my code in the `dispatch_asynch` function in `viewDidLoad`. Earlier, I wrote a function with the code to count issues and called that function in `dispatch_asynch`, which apparently doesn't work.

-- 2nd EDIT: Never mind that worked once but now it doesn't. Oh well.
